bin/lesson_check.py

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python
```

Comment:
`python` still refers to Python 2 on most platforms. Please revert this and all similar changes.

Reply:
Python 2 will not be supported from 2020. Reverting this to `python3` breaks this code in my system because I don't have such an alias. Since I am using a fresh install of Anaconda3, I assume others will have the same problem.

The correct path forward, if you wish to continue supporting Python 2, is to write a function that checks the version explicitly. I could do that if you prefer.

Reply:
> The correct path forward, if you wish to continue supporting Python 2, is to write a function that checks the version explicitly. I could do that if you prefer.

This script does not work with Python 2.

Reply:
This change would break thee script in Linux (and, hence, Travis and other CI platforms) and macOS.

Reply:
Perfect, then the more reason to add an explicit check at the beginning to check for that, instead of relying on an alias that is not present in all architectures/OSes. Using `python3` does not work on Windows 10 with a fresh installation of Anaconda 3. These are the executables I have in the PATH:

```
C:\Users\SoftwareCarpentry\Work\sc-teaching\intro-git-github $ python
python.exe     pythontex.exe  pythonw.exe
```

Reply:
Explicit check would not prevent the failure you're experiencing. 
Shebang would fail before the check can be implemented. We can replace the shebang with a note to use Python3 which might be `python` or `python3` depending on platform.

Reply:
The problem is that the setup makes use of `make` to call these python scripts and does so by executing them like this `./script.py`, which triggers the shebang. If I run the script manually (`python script.py`) indeed, I don't have this problem.

Maybe a better workaround is to modify the `Makefile` to allow people to edit which python is being used to call the scripts.

Reply:
We can detect which Python executable to use (python or python3) in a Makefile or in a new script that would then call this script.

Reply:
In general, I'm leaning towards removing the shebang rather than changing it (changing it might cause more troubles)

Reply:
Agreed. On a Mac, or on Linux, what interpreter shows up when you type `python`?

Reply:
Mac:
```sh
$ /usr/bin/python -V
Python 2.7.16
```

Reply:
On Ubuntu 16.04:
```
$ /usr/bin/python -V
Python 2.7.12
```

Even though I set my path with Anaconda so that the following works:
```
$ python
Python 3.6.4 |Anaconda, Inc.| (default, Jan 16 2018, 18:10:19) 
[GCC 7.2.0] on linux
```

Reply:
I deleted the shebang lines and added a `PYTHON` variable to the Makefile. It's defined as `python`, but it should be easier to modify to whatever executable we want. Does this work as a solution? See the latest commit to this PR.